### PR TITLE
Bugfixen - edge cases

### PR DESCRIPTION
**Pull request by qrohlf**
Handle a bunch of common edge cases

- show a message when the event has no routes instead of erroring
- handle all the possible permutations of `event.{starts,ends}_{on,at}`

This unfortunately bring is moment.js (14k gzipped) BUT I have plans to kill that when I extract & test `smartDate` after the soft launch (https://www.pivotaltracker.com/story/show/123916241)

Addresses
- https://www.pivotaltracker.com/story/show/123646471
- https://www.pivotaltracker.com/story/show/123646021
- https://www.pivotaltracker.com/story/show/123645961
- https://www.pivotaltracker.com/story/show/123646691
- https://www.pivotaltracker.com/story/show/123646691
- https://www.pivotaltracker.com/story/show/123648131

### CODE REVIEW

File: src/widgets/embed-configurator/EmbedConfigurator.js [hunk 1]
```diff
@@ -56,8 +56,8 @@ export default class EmbedConfigurator extends React.Component {
     if (iframeOpts.includeHot) {
       params.hot = hot
     }
-    let height = 530
-    if (sampleGraph) height = 530 + 150
+    let height = 550
```
Comment (qrohlf) [inline src/widgets/embed-configurator/EmbedConfigurator.js]: In the pathological case, the embed will increase by 18px in size when the viewport is way too narrow. This is a temporary solution - permanent fix planned for when we tackle embeds with > 4 routes via horizontal scrolling or stacking.

**Comment by qrohlf** [comment]
All the cases that event date code handles:

all_day_with_start
all_day_with_start

all_day_with_start_and_end
all_day_with_start_and_end

not_all_day_with_start
not_all_day_with_start

not_all_day_with_start_and_end
not_all_day_with_start_and_end

not_this_year
not_this_year

not_same_year
not_same_year

Note that since rails is converting into the event creator's timezone when it sends the JSON response, we _should_ be OK ignoring timezones for now in the JS parsing.

File: src/lib/smartDate.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  // if we're going to omit the year from the original dates,
+  const appendYear = sameYear && (sameYear !== moment().year())
+  const dateOpts = {includeYear: !sameYear, includeTime}
+  const rangeString = dates.map(d => formatSingleDate(d, dateOpts)).join(' to ')
+  if (appendYear) {
+    return `${rangeString}, ${sameYear}`
+  } else {
+    return rangeString
+  }
+}
```
Comment (qrohlf) [inline src/lib/smartDate.js]: Needs test, doesn't reaaaaly need moment, both are going to be fixed in https://www.pivotaltracker.com/story/show/123916241 after soft launch

**Comment by zackham** [comment]
nice on date handling, not worried about moment this moment, thanks for keeping pivotal in sync

**Pull request merged**